### PR TITLE
RD2: Prevent users from changing Elevate originated RDs

### PR DESCRIPTION
**Pull request by RadSFDeveloper**
Ensure that Recurring Donations created by the Payment Services integration are never allowed to be updated manually by a User in NPSP other than through the Payment Services Integration User, with the exception of rollup and custom fields, so that I can be sure that the Recurring Donation information in NPSP exactly matches what is in Elevate.

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata
Apex Class:
- PS_IntegrationServiceConfig
- PS_IntegrationServiceConfig_TEST

Custom Label:
- RD2_ElevatePermissionRequired

# Deleted Metadata

### CODE REVIEW

**Comment by RadSFDeveloper** [comment]
**lurch: attach W-040592

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000001HuLgQAK" target="_blank">W-040592</a>

File: src/labels/CustomLabels.labels [hunk 1]
```diff
@@ -1605,6 +1605,14 @@ If you&apos;re not sure how to resolve these errors, post a message in the Nonpr
         <shortDescription>Warning displayed on Customizable Rollups page when enhanced RDs are enabled</shortDescription>
         <value>&lt;b&gt;WARNING: You have Enhanced Recurring Donations enabled. If you disable Customizable Rollups, rollups to Recurring Donations will no longer function.&lt;/b&gt;</value>
     </labels>
+    <labels>
+        <fullName>RD2_ElevatePermissionRequired</fullName>
+        <categories>Recurring-Donations, Error</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Error when user does not have a permission to change an RD originated in Elevate</shortDescription>
+        <value>Recurring Donations from Elevate Payment Services cannot be modified directly in NPSP at this time.</value>
```
Comment (RadSFDeveloper) [inline src/labels/CustomLabels.labels:1614]: @sam-knox , can you please review the message?

Comment (sam-knox) [reply src/labels/CustomLabels.labels:1614]: @RadSFDeveloper - So in this case, if a user did want to make an update, they would need to make it through Elevate?

Comment (RadSFDeveloper) [reply src/labels/CustomLabels.labels:1614]: If the integration is enabled, user must have permissions in NPSP to change the Elevate RD. An integration user for now will have Elevate permissions for now (until bi-directional update is implemented).
@force2b , feel free to provide more detail if above is not correct.

Comment (sam-knox) [reply src/labels/CustomLabels.labels:1614]: Ok, thanks. Based on the ACs in the WI, how about this message:

"Recurring Donations from Elevate Payment Services can't be modified in NPSP except by the Elevate Payment Services Integration user."

@RadSFDeveloper @force2b

Comment (RadSFDeveloper) [reply src/labels/CustomLabels.labels]: The message has been updated.

**Pull request merged**